File: AnkerServices/src/main/java/com/ankercbt/ankerservices/service/AuthenticationService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package com.ankercbt.ankerservices.service;

/**
 *
 * @author tim
 */
public interface AuthenticationService {
    
    public boolean login(String identifier, String password);
    public boolean logout();
    public boolean isLoggedIn();
    
}

File: AnkerServices/src/main/webapp/js/app/util/security/AuthenticationService.js
/* 
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
define([
    'org.as/app'
], function(app) {
    app.factory('AuthenticationService', ['$http',
        function($http) {
            //local  vars
            return {
                user: {},
                tries: 0,
                login: function(user, password) {
                    var that = this;
                    return $http.post('login', {
                        user: user,
                        password: password
                    }).success(function(user) {
                        user.authenticated = true;
                        that.user = user;
                    }).error(function() {
                        that.tries++;
                        that.user = {
                            name: '',
                            roles: [],
                            authenticated: false
                        };
                    });
                },
                isAuthenticated: function() {
                    return this.authenticated;
                },
                logout: function() {
                    this.authenticated = false;
                }
            }
        }
    ]);

});

